src/components/Auth/Signup.js

```diff
@@ -0,0 +1,98 @@
+
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { setSignupInput } from '../../actions/userActions';
+
+export class Signup extends Component {
+  constructor() {
+    super();
+    this.state = {
+      username: '',
+      email: '',
+      password: ''
+    };
+    this.handleInput = this.handleInput.bind(this);
+  }
+
+  handleSubmit=(e) => {
+    e.preventDefault();
+    const { username, email, password } = this.state;
+    const data = {
+      username,
+      email,
+      password
+    };
+    const { setSignupInput } = this.props;
+    setSignupInput(data);
+  }
+
+  handleInput(e) {
+    this.setState({ [e.target.name]: e.target.value });
+  }
+
+
+  render() {
+    const { username, email, password } = this.state;
+    const { errors } = this.props;
+    return (
+      <div className="FormCenter">
+        <form onSubmit={this.handleSubmit} className="FormFields" autoComplete="off">
+          <div className="FormField">
+            <label className="FormField__Label" htmlFor="name">Full Name</label>
+            <input
+              type="text"
+              id="name"
+              className="FormField__Input"
+              placeholder="Enter your full name"
+              name="username"
+              value={username}
+              onChange={this.handleInput}
+            />
+          </div>
+          <div className="FormField">
+            <label className="FormField__Label" htmlFor="email">E-Mail Address</label>
+            <input
+              type="email"
+              id="email"
+              className="FormField__Input"
+              placeholder="Enter your email"
+              name="email"
+              value={email}
+              onChange={this.handleInput}
+            />
+          </div>
+          <div className="FormField">
+            <label className="FormField__Label" htmlFor="password">Password</label>
+            <input
+              type="password"
+              id="password"
+              className="FormField__Input"
+              placeholder="Enter your password"
+              name="password"
+              value={password}
+              onChange={this.handleInput}
+            />
+            { errors !== null && <small className="C-validationError">{errors.error}</small>}
+          </div>
+          <div className="FormField">
+            <button className="FormField__Button mr-20" type="submit">Sign Up</button>
+            <a href="/login" className="FormField__Link">I already have an account</a>
+          </div>
+        </form>
+      </div>
+    );
+  }
+}
+
```

Comment:
I like the fact that you have an `Input` component but the best thing would be reusing a form. Creating a form component that can be used on registration and login.

Reply:
Thank you for the feedback, I am going to consider this suggestion and refactor the code.